### ISSUE TITLE
MYSQL using ALT+X and PREPARE - EXECUTE

### ISSUE THREAD

**Issue by mlinsjr** [dbeaver/dbeaver#26769]
### Description

In the last version 2.4.3.20240421624 a new error is occorring in LINUX portuguese using a Mysql database.

When i execute step by step this instructions with CTRL+ENTER it works.

SET @a = 'Helllo ';
SET @b = 'world';
SET @s = CONCAT("SELECT '", @a , @b, "'");
SELECT @s;
PREPARE stmt3 FROM @s;
EXECUTE stmt3;

When i execute the all instructions with the command ALT+X,  i receive an error
 
Erro SQL [1064] [42000]: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'PREPARE stmt3 FROM @s;
EXECUTE stmt3;

### DBeaver Version

Comunity Edition 2.4.3.20240421624

### Operating System

LINUX UBUNTU portuguese VERSION="20.04.6 LTS (Focal Fossa)

### Database and driver

 Mysql 5.5.62-0+deb8u1

### Steps to reproduce

SET @a = 'Helllo ';
SET @b = 'world';
SET @s = CONCAT("SELECT '", @a , @b, "'");
SELECT @s;
PREPARE stmt3 FROM @s;
EXECUTE stmt3;

When i execute step by step this instructions with CTRL+ENTER it works.
When i execute the all instructions with the command ALT+X,  i receive an error.

**Comment by E1izabeth** [comment]
Thank you for report.
Could you try switching `Blank line is statement delimiter` setting to `Always`? Does it help?
![image](https://github.com/dbeaver/dbeaver/assets/28875055/be8457a0-d62c-4acb-ac2f-67ce518cf3f5)

**Comment by mlinsjr** [comment]
Bingo, I changed to **"always"** e and now is working 
Thanks!

**Comment by E1izabeth** [comment]
The fix will be included in DBeaver 24.0.4.
Also you can try Early Access version with fix https://dbeaver.io/files/ea/ now.

**Comment by cfischer00** [comment]
Not sure if this case is fixed. When using ALT-x on this script
```SQL
DROP TABLE IF EXISTS t;
CREATE TEMPORARY TABLE t (c text);
INSERT INTO t VALUES ('x');
-- comment here
```
the result is 2 rows in t.

If I remove the last comment, only 1 row. In execution log I can se that `INSERT INTO t VALUES ('x');` is executed twice.

If I change from smart to never as suggested: https://github.com/dbeaver/dbeaver/issues/26769#issuecomment-2077337343, the problem disappears.

I use version: 24.0.3.202404211624
